src/Microsoft.SqlTools.Hosting/Extensibility/ExtensionServiceProvider.cs

```diff
@@ -119,16 +119,108 @@ private void EnsureExtensionStoreRegistered<T>()
         /// <summary>
         /// Merges in new assemblies to the existing container configuration.
         /// </summary>
-        public void AddAssembliesToConfiguration(IEnumerable<Assembly> assemblies)
+        /// <typeparam name="T">Type of the service present in the assemblies</typeparam>
+        public void AddAssembliesToConfiguration<T>(IEnumerable<Assembly> assemblies)
         {
             Validate.IsNotNull(nameof(assemblies), assemblies);
             var previousConfig = config;
-            this.config = conventions => {
+            this.config = conventions =>
+            {
                 // Chain in the existing configuration function's result, then include additional
                 // assemblies
                 ContainerConfiguration containerConfig = previousConfig(conventions);
                 return containerConfig.WithAssemblies(assemblies, conventions);
             };
+            ExtensionStore store = new ExtensionStore(typeof(T), config);
+
+            // If the service type is already registered, replace the existing registration with the new one
+            if (this.services.ContainsKey(typeof(T)))
+            {
+                this.services[typeof(T)] = () => store.GetExports<T>();
+            }
+            else
+            {
+                base.Register<T>(() => store.GetExports<T>());
+            }
+        }
+
+        /// <summary>
+        /// Creates a service provider by loading a set of named assemblies, expected to be <paramref name="directory"/>
+        /// </summary>
+        /// <param name="directory">Directory to search for included assemblies</param>
+        /// <param name="inclusionList">full DLL names, case insensitive, of assemblies to include</param>
+        /// <returns><see cref="ExtensionServiceProvider"/> instance</returns>
+        public static ExtensionServiceProvider CreateFromAssembliesInDirectory(string directory, IList<string> inclusionList)
+        {
+            //AssemblyLoadContext context = new AssemblyLoader(directory);
+            var assemblyPaths = Directory.GetFiles(directory, "*.dll", SearchOption.TopDirectoryOnly);

```

Comment:
A good follow up task would be to add more debug logging to this startup stuff (for example "Loading service assemblies from <directory>...") to help track down issues later on.